hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/operations/HederaDelegateCallOperation.java

```diff
@@ -55,16 +34,26 @@
 public class HederaDelegateCallOperation extends DelegateCallOperation {
 
     private final BiPredicate<Address, MessageFrame> addressValidator;
+    private final Predicate<Address> systemAccountDetector;
 
     public HederaDelegateCallOperation(
-            GasCalculator gasCalculator, BiPredicate<Address, MessageFrame> addressValidator) {
+            GasCalculator gasCalculator,
+            BiPredicate<Address, MessageFrame> addressValidator,
+            Predicate<Address> systemAccountDetector) {
         super(gasCalculator);
         this.addressValidator = addressValidator;
+        this.systemAccountDetector = systemAccountDetector;
     }
 
     @Override
     public OperationResult execute(MessageFrame frame, EVM evm) {
         return HederaEvmOperationsUtil.addressCheckExecution(
```

Comment:
This is one of the places where I feel that our obsession with "duplicate line detection" has had a negative impact on our code quality.  Instead of writing a readable segment of code we parse it apart into several incomprehensible steps and ask a utlity method to make sense of it.  Now we are adding in "do this if it is system" because a couple of operations need it.  Now every operation has to do it.  

I feel our sonar obsession has actively harmed the maintainability of the code.